The job **`x86_64-gnu-tools`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/10462712773/job/28973449676) [(plain)](https://github.com/rust-lang/rust/commit/3559a7fed2cacfc64ef3d09c3b61f1567ca74d8a/checks/28973449676/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
.......... (120/129)
........F  (129/129)


/checkout/tests/rustdoc-gui/target.goml target... FAILED
[ERROR] `tests/rustdoc-gui/target.goml` line 14
    from `tests/rustdoc-gui/target.goml` line 21: The following errors happened (for selector `#method\.a_method:target`): [expected `rgba(255, 236, 164, 0.06)` for key `background-color`, found `rgba(255, 186, 86, 0.337)`]: for command `assert-css: ("#method\.a_method:target", {
            "background-color": |background|,
            "border-right": "3px solid " + |border|,
        })`
[ERROR] `tests/rustdoc-gui/target.goml` line 14
    from `tests/rustdoc-gui/target.goml` line 26: The following errors happened (for selector `#method\.a_method:target`): [expected `#494a3d` for key `background-color`, found `#524d39`]: for command `assert-css: ("#method\.a_method:target", {
            "background-color": |background|,
            "border-right": "3px solid " + |border|,
        })`
Error: ()
Build completed unsuccessfully in 0:03:09
  local time: Tue Aug 20 00:41:19 UTC 2024
  network time: Tue, 20 Aug 2024 00:41:19 GMT
```

</details>

--- a/src/librustdoc/html/static/css/rustdoc.css
+++ b/src/librustdoc/html/static/css/rustdoc.css
@@ -1547,10 +1547,23 @@ instead, we check that it's not a "finger" cursor.
 	margin-left: 24px;
 }
 
+@keyframes targetfadein {
+	from {
+		background-color: var(--main-background-color);
+	}
+	10% {
+		background-color: var(--target-border-color);
+	}
+	to {
+		background-color: var(--target-background-color);
+	}
+}
+
 :target {
 	padding-right: 3px;
 	background-color: var(--target-background-color);
 	border-right: 3px solid var(--target-border-color);
+	animation: 0.65s cubic-bezier(0, 0, 0.1, 1.0) 0.1s targetfadein;
 }
 
 .code-header a.tooltip {

--- a/tests/rustdoc-gui/target.goml
+++ b/tests/rustdoc-gui/target.goml
@@ -11,7 +11,7 @@ define-function: (
     [theme, background, border],
     block {
         call-function: ("switch-theme", {"theme": |theme|})
-        assert-css: ("#method\.a_method:target", {
+        wait-for-css: ("#method\.a_method:target", {
             "background-color": |background|,
             "border-right": "3px solid " + |border|,
         })

